databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java

```diff
@@ -245,15 +312,19 @@ private Response executeInner(Request in, String path) {
       Response out = null;
 
       // Authenticate the request. Failures should not be retried.
-      in.withHeaders(config.authenticate());
+      in.withHeaders(authenticateFunc.apply(null));
```

Comment:
Suggestion: we can treat `authenticateFunc` as nullable to indicate whether headers need to be added. Thus:
```java
if (authenticateFunc != null) {
    in.withHeaders(authenticateFunc.apply(null));
}
```
This way, users don't have to supply an empty authenticateFunc in the builder.